homeassistant/components/evse_wifi/__init__.py

```diff
@@ -0,0 +1,41 @@
+"""The EVSE Wifi integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_HOST, CONF_INTERVAL, CONF_MAX_CURRENT, CONF_NAME, DOMAIN
+from .evse import EVSE
+
+PLATFORMS: list[Platform] = [
+    Platform.SENSOR,
+    Platform.NUMBER,
+    Platform.BUTTON,
+    Platform.SWITCH,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up EVSE Wifi from a config entry."""
+
+    evse = EVSE(
+        name=entry.data[CONF_NAME],
+        host=entry.data[CONF_HOST],
+        max_current=entry.data[CONF_MAX_CURRENT],
+        interval=entry.data[CONF_INTERVAL],
+    )
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = evse
```

Comment:
We only do this after we have endured that we can reach the api and we can authenticate